src/discord_utils.py

```diff
@@ -31,6 +31,26 @@ async def try_set_pin(message: Message, pin_state: bool) -> None:
         print("Altering message pin state failed:", e)
 
 
+def attachment_local_filepath(message_id: int, attachment: Attachment) -> str:
```

Comment:
nit: I would call this `build_filepath_for_attachment`. `attachment_local_filepath` sounds like we're extracting a filepath that already exists.